Sources/Components/IconCollection/HorizontalIconCollectionViewCell.swift

```diff
@@ -0,0 +1,116 @@
+//
+//  Copyright © FINN.no AS, Inc. All rights reserved.
+//
+
+import UIKit
+
+public class HorizontalIconCollectionViewCell: UICollectionViewCell {
+    private static let titleSideMargin = CGFloat.mediumSpacing
+
+    static func height(for viewModel: IconCollectionViewModel, withWidth width: CGFloat) -> CGFloat {
+        let imageSize = viewModel.image.size
+        let textWidth = width - imageSize.width - (2 * titleSideMargin)
+
+        let textRect = viewModel.text.boundingRect(
+            with: CGSize(width: textWidth, height: CGFloat.greatestFiniteMagnitude),
+            options: .usesLineFragmentOrigin,
+            attributes: [.font: UIFont.body],
+            context: nil
+        )
+
+        guard let title = viewModel.title else {
+            return max(imageSize.height, textRect.height)
+        }
+
+        let titleRect = title.boundingRect(
+            with: CGSize(width: textWidth, height: CGFloat.greatestFiniteMagnitude),
+            options: .usesLineFragmentOrigin,
+            attributes: [.font: UIFont.bodyStrong],
+            context: nil
+        )
+
+        let textHeight = titleRect.height + textRect.height + .smallSpacing
+
+        return max(imageSize.height, textHeight)
+    }
+
+    private lazy var iconImageView: UIImageView = {
+        let imageView = UIImageView(withAutoLayout: true)
+        imageView.setContentCompressionResistancePriority(.defaultHigh, for: .horizontal)
+        imageView.setContentHuggingPriority(.defaultHigh, for: .horizontal)
+        imageView.contentMode = .scaleAspectFit
+        return imageView
+    }()
+
+    private lazy var titleLabel: UILabel = {
+        let label = UILabel(withAutoLayout: true)
+        label.font = .body
+        label.textColor = .textPrimary
+        label.adjustsFontSizeToFitWidth = true
+        label.numberOfLines = 0
+        return label
+    }()
+
+    private lazy var bodyLabel: UILabel = {
+        let label = UILabel(withAutoLayout: true)
+        label.font = .bodyStrong
+        label.textColor = .textPrimary
+        label.adjustsFontSizeToFitWidth = true
+        label.numberOfLines = 0
```

Comment:
are these two lines conflicting? (58-59)
- number of lines = 0 => use as many lines as you need
- adjustsFontSizeToWidth => if you don't have space shrink the font size

also, why not a use a Label instead?

Reply:
Yup, they're definitely conflicting. Removing `adjustsFontSizeToFitWidth` and using `Label` instead 👍